crates/hyperswitch_connectors/src/connectors/santander.rs

```diff
@@ -644,25 +753,34 @@ impl ConnectorIntegration<Execute, RefundsData, RefundsResponseData> for Santand
         req: &RefundsRouterData<Execute>,
         connectors: &Connectors,
     ) -> CustomResult<String, errors::ConnectorError> {
-        let end_to_end_id = req
-            .request
-            .connector_metadata
-            .as_ref()
-            .and_then(|metadata| metadata.get("end_to_end_id"))
-            .and_then(|val| val.as_str().map(|id| id.to_string()))
-            .ok_or_else(|| errors::ConnectorError::MissingRequiredField {
-                field_name: "end_to_end_id",
-            })?;
-
-        let refund_id = req.request.connector_refund_id.clone();
-        Ok(format!(
-            "{}{}{}{}{:?}",
-            self.base_url(connectors),
-            "pix/",
-            end_to_end_id,
-            "/refund/",
-            refund_id
-        ))
+        match req.payment_method {
+            enums::PaymentMethod::BankTransfer => {
```

Comment:
We could consider matching on req.request.payment_method first and then on req.request.payment_method_type. That could make the logic a bit more explicit - you could  make it the same for the other flows too.